proto/accessgraph/v1/query.proto

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package accessgraph.v1;
+
+option go_package = "github.com/gravitational/teleport/gen/proto/go/accessgraph/v1;accessgraphv1";
+
+// QueryRequest is a request to query the access graph.
+message QueryRequest {
+  string query = 1;
+}
+
+// Node is a node in the access graph.
+message Node {
+  string id = 1;
+  string kind = 2;
+  string sub_kind = 3;
+  string name = 4;
+  map<string, string> labels = 5;
+  string hostname = 6;
+}
+
+// Edge is an edge in the access graph.
+message Edge {
+  string from = 1;
+  string to = 2;
+  string type = 3;
+}
+
+// QueryResponse is a response to a query.
+message QueryResponse {
+  repeated Node nodes = 1;
```

Comment:
It's difficult to understand how a response could hold multiple nodes and edges without knowing what a query might be.

Reply:
The query language is SQL-ish, and the underlying response is SQL-ish as well (tabular). However, the UI (which is the ultimate client of QueryService) does not care about the tabular format in particular: it wants to render the selected (according to the query) subgraph of the entire access graph. Hence, nodes and edges are enough. Not sure why the query would need to be returned as well?

Reply:
Just to clarify I'm not asking for the query to be returned, I'm just saying that it's difficult for a reader to understand the response without any explanation about what the input is.